packages/design-system/src/components/text-field.tsx

```diff
@@ -147,13 +147,14 @@ const TextFieldBase = styled("div", {
 });
 
 type TextFieldProps = Omit<
-  React.ComponentProps<typeof TextFieldBase> & React.ComponentProps<"input">,
-  "prefix"
-> & {
-  inputRef?: React.Ref<HTMLInputElement>;
-  prefix?: React.ReactNode;
-  suffix?: React.ReactNode;
-};
+  React.ComponentProps<typeof TextFieldBase>,
+  "onChange" | "onFocus" | "onBlur" | "prefix"
+> &
+  React.ComponentProps<"input"> & {
```

Comment:
I wonder if something like this would be even better:

```suggestion
  React.ComponentProps<typeof InputBase>,
  "prefix"
> & {
```

BTW still not sure why `prefix` needs to be omitted but it looks like this is this `prefix`: https://yeahexp.com/what-is-the-prefix-attribute-for-in-the-html-tag/